### ISSUE TITLE
Add donation_url to `local_site` table.

### ISSUE THREAD

**Issue by Kap2022** [LemmyNet/lemmy#2332]
The heart icon in the header of all default installs of lemmy instances that clicks through to the Support Lemmy page https://join-lemmy.org/donate only provides the option for people to support the developers of lemmy - which is important. But there are other cost involved in running an instance that need to be covered such as hosting, content creation, moderation.

As a result I have seen some instances swap out the https://join-lemmy.org/donate link for another one that sends the donations to the instance and nothing for the developers.

I think neither of these situations is ideal and each is unfair to the other.
-->

Add settings to a lemmy instance settings page where we can specify crypto wallet addresses to cover hosting, content creation, moderation etc.

Then rather then the heart icon clicking through directly to https://join-lemmy.org/donate, insert a page that gives the user to choose to donate to development or hosting, content creation, moderation and insert the wallet addresses specified in the settings page.

This seems fair as it gives all parties the opportunities to collect donations and makes it less likely instances will swap out the link for one that excludes developers all together.

**Comment by dessalines** [comment]
You can already use your site sidebar, which lets you use markdown links, to put whatever instance donation links you want.

**Comment by silasabbott** [comment]
I'd like to counter this. The sidebar is not easily parsed and consequently makes it far less likely for donation information to be displayed in a lot of circumstances, especially on mobile interfaces.

Having one field with an array of donations links, or even just a single link would really help third-party apps display donation information.

**Comment by dessalines** [comment]
Still not sure about this... looking at a few servers rn, their donation section of the sidebar used a markdown description anyway, and none of them are the same. They use different description lines, sometimes one or multiple links, sometimes images... etc. Moving it to a field removes that flexibility, or makes it possibly redundant.

I spose I'd need some good cases for, where in apps you'd want to display donation links only, but not the rest of the sidebar.

**Comment by silasabbott** [comment]
Ah, that makes sense. The main issue I see with having donation info in the sidebar is that most mobile clients do not show the sidebar prominently (or at all) in their apps. It's such a large block of content that it's usually hidden away in a menu somewhere. Maybe there could be a single field for the donation link and a separate field for donation notes? I do think sacrificing a bit of flexibility here to make donations more accessible would be alright.

Here are some further ways this could be used:

1. Clients could show server donation links next to their own donation link on a page, view, or dialog. Currently this isn't possible without figuring out a way to parse the sidebar and find the right link(s).
2. Clients could integrate server donations into their own donation flow (by link, embed, or api). Again, you wouldn't parse or show the whole server sidebar here.
3. Clients could pull specific info from Ko-Fi or other platforms to show monthly goals and more.

Lemmy's financial hurdles could really make or break it in the long run, that's why I'm pushing for this. Incoming users from big social media need to understand that Lemmy is self-funded and runs off of donations. The more accessible these details are to developers, the more likely they are to be shown to Lemmy users.

**Comment by dessalines** [comment]
Makes sense. I'd be okay with this, as long as its a simple single URL field.

**Comment by dessalines** [comment]
No one has wanted to work on this, and the sidebar is much more flexible for this purpoes. I'll re-open if someone wants to work on it.
